### ISSUE TITLE
Show term name in only-one-level error 

### ISSUE THREAD

**Issue by jtrakk** [JuliaStats/StatsModels.jl#229]
```jl
ArgumentError: only one level found: foo (need at least two to compute contrasts).
Stacktrace:
  [1] StatsModels.ContrastsMatrix(contrasts::DummyCoding, levels::Vector{Int64})
    @ StatsModels ~/.julia/packages/StatsModels/MDeyQ/src/contrasts.jl:185
```

it would be helpful if it would say which column/term the problem is in.

**Comment by kleinschmidt** [comment]
I agree that would be nice to have.  It's currently a bit tricky since you'll see that it's the `ContrastsMatrix` constructor that does that check and throws the error, and that doesn't know about the context in which it's being called (including the term name).  So we'd have to catch and re-throw the error at the term construction site (probably `concrete_term` if memory serves) but that seems a bit icky to me somehow.

**Comment by kleinschmidt** [comment]
(I will say that the idea with printing the actual level it did find is to help track down which column it is that caused the problem, even though giving the column name would be even better)
